File: templates/julia/bot.jl
using Sockets

function readview(sock)
	view = readline(sock)
	view == "" && return
	size = length(view) - 1
	view *= "\n"
	for i = 1:size
		line = readline(sock) * "\n"
		view = view * line
	end
	return view
end

sock = connect(get(ARGS, 1, "localhost"), get(ARGS, 2, 63187))
while isopen(sock)
	view = readview(sock)
	view == "" && break
	print(view, "Command (q<>^v): ")
	cmd = read(stdin, Char)
	print("\n")
	if cmd == 'q'
		break
	elseif cmd == '\n'
		cmd = '^'
	end
	write(sock, cmd)
end
close(sock)

File: templates/python/bot.py
#!/usr/bin/env python

import sys
import socket


def read_view(f):
    view = f.readline()
    if not view:
        return
    for x in range(2, len(view)):
        line = f.readline()
        if not line:
            return
        view += line
    return view


def main(host='localhost', port=63187):
    s = socket.socket()
    s.connect((host, port))
    f = s.makefile()
    while True:
        try:
            view = read_view(f)
            if not view:
                break
            sys.stdout.write(view + "Command (q<>^v): ")
            cmd = sys.stdin.readline()[0]
            if cmd == 'q':
                break
            if cmd == '\n':
                cmd = '^'
            s.send(cmd if sys.version_info[0] < 3 else str.encoded(cmd))
        except Exception as e:
            print(e)
            break
    s.close()


if __name__ == '__main__':
    main( * sys.argv[1:])
